WordPress/src/main/java/org/wordpress/android/ui/stockmedia/StockMediaPickerActivity.java

```diff
@@ -0,0 +1,590 @@
+package org.wordpress.android.ui.stockmedia;
+
+import android.app.FragmentManager;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SearchView;
+import android.support.v7.widget.Toolbar;
+import android.text.Html;
+import android.text.TextUtils;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.StockMediaActionBuilder;
+import org.wordpress.android.fluxc.model.SiteModel;
+import org.wordpress.android.fluxc.model.StockMediaModel;
+import org.wordpress.android.fluxc.store.StockMediaStore;
+import org.wordpress.android.ui.media.MediaPreviewActivity;
+import org.wordpress.android.util.ActivityUtils;
+import org.wordpress.android.util.AniUtils;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.DisplayUtils;
+import org.wordpress.android.util.NetworkUtils;
+import org.wordpress.android.util.PhotonUtils;
+import org.wordpress.android.util.StringUtils;
+import org.wordpress.android.util.ToastUtils;
+import org.wordpress.android.util.WPLinkMovementMethod;
+import org.wordpress.android.widgets.WPNetworkImageView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class StockMediaPickerActivity extends AppCompatActivity implements SearchView.OnQueryTextListener {
+    private static final int MIN_SEARCH_QUERY_SIZE = 3;
+    private static final String TAG_RETAINED_FRAGMENT = "retained_fragment";
+
+    private static final String KEY_SEARCH_QUERY = "search_query";
+    private static final String KEY_CAN_LOAD_MORE = "can_load_more";
+    private static final String KEY_NEXT_PAGE = "next_page";
+
+    private SiteModel mSite;
+
+    private StockMediaAdapter mAdapter;
+    private StockMediaRetainedFragment mRetainedFragment;
+
+    private RecyclerView mRecycler;
+    private ViewGroup mSelectionBar;
+    private TextView mTextAdd;
+
+    private SearchView mSearchView;
+    private String mSearchQuery;
+    private final Handler mHandler = new Handler();
+
+    private int mThumbWidth;
+    private int mThumbHeight;
+
+    private boolean mIsFetching;
+    private boolean mCanLoadMore;
+    private int mNextPage;
+
+    @Inject Dispatcher mDispatcher;
+    @Inject StockMediaStore mStockMediaStore;
```

Comment:
It might be worth adding a `@SuppressWarnings("unused")` to this property since it's actually required to receive the `onStockMedia` events afaik.